src/app/components/store-products/store-products.component.ts

```diff
@@ -8,4 +19,67 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class StoreProductsComponent {
+  private _keyWordForSearchSubject: BehaviorSubject<string> = new BehaviorSubject<string>('');
+  public keyWordForSearch$: Observable<string> = this._keyWordForSearchSubject.asObservable();
+
+  readonly storeId$: Observable<string> = this._activatedRoute.params.pipe(
+    map((params) => params['storeId']),
+    shareReplay(1)
+  );
+  readonly tagsMap$: Observable<Record<string, TagModel>> = this._tagsService.getTagsMap();
+
+  readonly storeDetails$: Observable<StoreQueryModel> = combineLatest([this.storeId$, this.tagsMap$]).pipe(
+    switchMap(([storeId, tagsMap]) =>
+      this._storesService.getSingleStoreById(storeId).pipe(map((store) => this.mapToStoreQueryModel(tagsMap, store)))
+    )
+  );
+
+  readonly productsInStoreList$: Observable<ProductModel[]> = combineLatest([
+    this.keyWordForSearch$,
+    this.storeId$
+  ]).pipe(
+    switchMap(([keyWord, storeId]) =>
+      this._productsService
+        .getAllProducts()
+        .pipe(
+          map((products) =>
+            products.filter((product) =>
+              product.storeIds.some(
+                (productStoreId) => productStoreId === storeId && product.name.toLowerCase().includes(keyWord)
+              )
+            )
+          )
+        )
+    )
+  );
+
+  readonly storeSearch: FormGroup = new FormGroup({ keyWord: new FormControl() });
+
+  constructor(
+    private _activatedRoute: ActivatedRoute,
+    private _storesService: StoresService,
+    private _productsService: ProductsService,
+    private _tagsService: TagsService
+  ) {}
+
+  mapToStoreQueryModel(tagsMap: Record<string, TagModel>, store: StoreModel): StoreQueryModel {
+    return {
+      id: store.id,
+      name: store.name,
+      logoUrl: store.logoUrl,
+      distanceInKm: Math.round(store.distanceInMeters / 100) / 10,
+      tags: store.tagIds.map((tagId) => tagsMap[tagId]?.name)
+    };
+  }
+
+  onStoreSearchSubmitted(storeSearch: FormGroup): void {
+    //for debugging
+    console.log('Searching for products with keyWord: ' + storeSearch.value.keyWord);
+    this._keyWordForSearchSubject.next(storeSearch.value.keyWord.toLowerCase());
```

Comment:
try to use valueChanges instead of a submit

Reply:
I wanted to make it work on pressing Enter or clicking on the icon. But shure I can also do it on value changes :)

Reply:
Search logic changed :)

Reply:
ok i see, also good idea